scripts/check_pg_extensions.py

```diff
@@ -0,0 +1,108 @@
+"""
+    Usage as a script:
+
+    (while cf oauth-token && sleep 2m ; do : ; done) | python check_pg_extensions.py https://my.api.base.url.example.com service_guid extension_sets.csv
+
+    This script generally takes longer to run than a single oauth token lasts
+    so it continually polls for new tokens being fed in to its stdin, which
+    can be provided by a looping `cf oauth-token` running every few minutes.
+
+    The extension_sets csv is expected to have one set of extension names
+    per row, one extension per column, in expected order of insertion.
+    Really it's used as more of a list of lists than a table.
+
+    Depends on requests. Not fussy about version.
+"""
+
+import csv
+import json
+from select import select
+import sys
+from time import sleep
+from urllib.parse import urljoin
+
+import requests
+
+
+_auth_header = None
+def _get_auth_header(f=sys.stdin):
+    global _auth_header
+    if _auth_header is None or select([f], [], [], 0)[0]:
+        while True:
+            line = next(f).strip()
+            if line:
+                print("<new auth token>", end="")
+                sys.stdout.flush()
+                _auth_header = {
+                    "Authorization": line,
+                }
+
+            if _auth_header is not None:
+                break
+
+    return _auth_header
+
+
+def _wait_for_success(api_base_url, service_guid):
+    while True:
+        sleep(10)
```

Comment:
Could you use a non-fixed delay on this? Fixed delays often seem to be the wrong solution, in my opinion. Could we usefully try linear or exponential backoff, eg

```suggestion
    delay = 1
    while True:
        sleep(delay)
        delay = delay * 2
```

Reply:
We _could_, but the problem is this script is slow enough as it is, and I quite like the reassurance that as soon as the server side is ready for the next step we will pick it up within 10 seconds. With a backoff it would potentially be wasting a couple of minutes by the time it had been waiting 5 minutes already.

Backoffs are particularly useful when you have multiple clients potentially hammering something, but in this case i can't imagine more than one person running this script at a time.

(this is not supposed to be a perfect script btw, it's mostly just me committing what I used so it doesn't get lost)